### PR TITLE
fix: harden launchd plist paths and maintenance resume

### PR DESCRIPTION
**Pull request by EtanHey**
## Summary
- Render Homebrew-installed launchd paths through the stable `opt/brainlayer` symlink instead of versioned `Cellar/brainlayer/<version>` paths.
- Make launchd service loading idempotent as `bootout -> enable -> bootstrap -> print`, with multi-service installer targets attempting remaining services after one service fails.
- Make maintenance resume every paused service before reporting resume failures, including exception notes when maintenance work fails first.
- Add regression coverage for Homebrew Cellar path leakage, bootstrap-error continuation, preflight failure short-circuiting, and maintenance resume aggregation.

## Test plan
- RED first: new BL-2/BL-9 tests failed on current behavior before production edits.
- `pytest tests/test_installable_build.py::test_launchd_installer_renders_homebrew_opt_symlink_instead_of_cellar_version tests/test_installable_build.py::test_launchd_installer_attempts_remaining_services_after_bootstrap_error tests/test_maintenance_routine.py::test_maintenance_resume_attempts_all_services_after_mid_resume_failure` -> failed before fix, passed after fix.
- Reviewer REDs: `pytest tests/test_installable_build.py::test_launchd_installer_does_not_load_services_when_env_runner_install_fails tests/test_maintenance_routine.py::test_maintenance_body_error_reports_resume_failures_as_exception_note` -> failed before review fixes, passed after review fixes.
- `pytest tests/test_installable_build.py tests/test_maintenance_routine.py tests/test_launchd_hygiene.py` -> 58 passed, 1 warning.
- `ruff check src/brainlayer/maintenance.py tests/test_installable_build.py tests/test_maintenance_routine.py` -> passed.
- `bash -n scripts/launchd/install.sh` -> passed.
- `git diff --check` -> passed.
- Full local `pytest` on Python 3.13 -> 3262 passed, 49 skipped, 5 xfailed, 329 warnings in 448.41s.
- Pre-push gate on Python 3.12 -> pytest unit suite 3161 passed, 9 skipped, 61 deselected, 1 xfailed; MCP registration 3 passed; isolated eval/hook routing 40 passed; bun test 1 passed; FTS5 regression shell passed.

## Review notes
- Local `coderabbit review --agent` was run under a 180s hard timeout and timed out while still in `reviewing`; not treated as clean.
- Spawned reviewer pane `brainlayerCodex-pending-1782421332-4r6u` returned FAIL with two findings. Both were fixed with RED tests first:
  - installer preflight failures could be ignored under `if ! install_plist` because Bash disables `errexit` in conditional function calls.
  - maintenance body errors did not report resume failures. Now resume failures are attached as exception notes when a body error is already propagating.
- Full `ruff check .` was also run and still fails on unrelated pre-existing lint in `scripts/*.py` outside this diff; scoped ruff on touched Python/test files passes.

## Tracking
- Lifecycle status appended to `collab/2026-06-25-brainlayer-worker-status.md` for STARTED and GREEN.

---

> [!NOTE]
> **Medium Risk**
> Changes affect local launchd agent installation and maintenance shutdown/resume of core background services; misbehavior could leave agents down or mis-pointed after Homebrew upgrades, though behavior is heavily regression-tested.
> 
> **Overview**
> **Launchd installer** now rewrites Homebrew `Cellar/brainlayer/<version>` paths to stable `opt/brainlayer` targets in rendered plists, and **`load_plist`** handles missing-service `enable` by retrying after bootstrap with explicit failures on other steps. Preflight and render steps propagate errors correctly (including under `if ! install_plist`), **`install_many`** keeps going when one service fails, and the **`all`** target only drops legacy **`enrich`** after **`enrichment`** loads successfully.
> 
> **Maintenance** resumes every quiesced launchd service even when one resume fails, then raises **`MaintenanceAbort`** with aggregated resume details; if maintenance work already failed, those resume failures are appended to the exception reason and notes.
> 
> Regression tests cover opt-vs-Cellar rendering, bootstrap/enable edge cases, preflight short-circuiting, and resume aggregation.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 641dbe62a4475daa2a7004d1ca07c8dfb8737870. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Harden launchd plist installer paths and maintenance resume error handling
> - Adds `stable_brainlayer_path` to [install.sh](https://github.com/EtanHey/brainlayer/pull/552/files#diff-c2e9ec925e37d2fc11ea258b25bb7f65308a6755174d1b556a6c077d798bf756) to normalize Homebrew Cellar paths to stable `opt` symlinks, preventing versioned Cellar paths from being embedded in rendered plists.
> - Hardens `install_plist`, `load_plist`, `install_backup_script`, and related handlers with `|| return 1` guards to abort on failure and prevent partial installs or spurious `launchctl` calls.
> - Adds `install_many` batch installer that continues past individual service failures and reports aggregate results; the `maintenance` command now uses it for nightly and weekly services.
> - Changes `_resume_services` in [maintenance.py](https://github.com/EtanHey/brainlayer/pull/552/files#diff-5623eb0b0ddbc791b2b25c828a74927787f1cf710501ff90555ebf92981b041d) to attempt all services before reporting failures, and `run_maintenance` now raises `MaintenanceAbort` if any resume fails.
> - Behavioral Change: `load_plist` now retries `launchctl enable` after bootstrap if the service is not found before bootstrap, rather than failing immediately.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 641dbe6.</sup>
> 
>

### CODE REVIEW

File: scripts/launchd/install.sh [hunk 1]
```diff
@@ -155,7 +193,26 @@ install_plist() {
     echo "Installed: $dst"
     echo "  Logs: $LOG_DIR/ and $BRAINLAYER_LOG_DIR/"
 
-    load_plist "$name"
+    if ! load_plist "$name"; then
+        return 1
+    fi
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh:212]: Fixed in 522cc7c: plist/env-runner/render helper failures now return explicitly even when called under install_many, preserving continue-on-error per service without swallowing setup failures.

File: scripts/launchd/install.sh [hunk 2]
```diff
@@ -264,23 +334,33 @@ case "${1:-all}" in
...
+        elif ! install_many jsonl-backup; then
+            failures=1
+        fi
+        if ! install_many maintenance-nightly maintenance-weekly health-check; then
+            failures=1
+        fi
+        # Remove old enrich plist only after the replacement enrichment batch loads.
+        if [ "$main_services_ok" -eq 1 ]; then
+            remove_plist enrich 2>/dev/null || true
+        fi
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh]: Fixed in 4c66265: the all path now tracks enrichment_ok independently. It removes legacy enrich after replacement enrichment loads, even if a later sibling service such as repair-fts fails, while still exiting nonzero for that sibling failure.

File: scripts/launchd/install.sh [hunk 3]
```diff
@@ -264,23 +320,28 @@ case "${1:-all}" in
...
+            failures=1
+        elif ! install_many jsonl-backup; then
+            failures=1
+        fi
+        if ! install_many maintenance-nightly maintenance-weekly health-check; then
+            failures=1
+        fi
         # Remove old enrich plist if present
         remove_plist enrich 2>/dev/null || true
+        if [ "$failures" -ne 0 ]; then
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh:367]: Fixed in 522cc7c and refined in 4c66265: legacy enrich is preserved if replacement enrichment fails, but removed once replacement enrichment itself loads successfully.

File: src/brainlayer/maintenance.py [hunk 4]
```diff
@@ -528,8 +542,16 @@ def run_maintenance(mode: str, *, config: MaintenanceConfig | None = None, dry_r
...
                 now=config.now_fn(),
             )
+    except BaseException as exc:
+        body_error = exc
+        raise
     finally:
-        _resume_services(config.repo_root, services)
+        resume_failures = _resume_services(config.repo_root, services)
+        if resume_failures and body_error is not None:
+            body_error.add_note(_format_resume_failures(resume_failures))
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/maintenance.py]: Fixed in 522cc7c: resume failures during a MaintenanceAbort are now included in MaintenanceAbort.reason/args as well as exception notes, so the CLI abort payload is actionable.

File: scripts/launchd/install.sh [hunk 5]
```diff
@@ -264,23 +320,28 @@ case "${1:-all}" in
...
-        install_jsonl_backup_script
-        install_plist jsonl-backup
-        install_plist maintenance-nightly
-        install_plist maintenance-weekly
-        install_plist health-check
+        failures=0
+        if ! install_many index drain watch hotlane-brainbar enrichment decay wal-checkpoint repair-fts; then
+            failures=1
+        fi
+        if ! install_backup_script; then
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh:350]: Fixed in 522cc7c: wrapper render failures are preserved with explicit return checks, and test coverage verifies backup/jsonl plists are not bootstrapped after failed wrapper rendering.

File: src/brainlayer/maintenance.py [hunk 6]
```diff
@@ -508,6 +520,8 @@ def run_maintenance(mode: str, *, config: MaintenanceConfig | None = None, dry_r
     services = DEFAULT_SERVICES
     if mode == "burn":
         services = (*REFEED_SERVICES, "drain")
+    resume_failures: list[tuple[str, Exception]] = []
+    body_error: BaseException | None = None
     _quiesce_services(services)
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/maintenance.py:525]: Fixed in 522cc7c: when maintenance body failure and resume failure both occur, resume failures are added to the exception note and to MaintenanceAbort.reason/args so CLI JSON surfaces them.

File: scripts/launchd/install.sh [hunk 7]
```diff
@@ -132,11 +170,11 @@ install_plist() {
         return 1
     fi
 
-    install_env_runner
-    verify_config_file
+    install_env_runner || return 1
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh:187]: Fixed in 522cc7c: install_env_runner now uses install -m 0755 with an explicit || return 1, and regression coverage asserts services are not bootstrapped when the env-runner copy fails.

File: scripts/launchd/install.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh:350]: Fixed in 522cc7c: backup and JSONL wrapper render/chmod steps now use explicit || return 1 checks, and regression coverage asserts backup jobs are not loaded when wrapper rendering fails.

File: scripts/launchd/install.sh [hunk 8]
```diff
@@ -108,10 +137,19 @@ load_plist() {
...
-    launchctl bootstrap "gui/$UID" "$dst"
-    launchctl print "gui/$UID/$label" >/dev/null
+    if ! launchctl enable "gui/$UID/$label"; then
+        echo "ERROR: launchctl enable failed for $label" >&2
+        return 1
+    fi
+    if ! launchctl bootstrap "gui/$UID" "$dst"; then
+        echo "ERROR: launchctl bootstrap failed for $label" >&2
+        return 1
+    fi
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply scripts/launchd/install.sh]: Fixed in 522cc7c: launchd enable now treats missing pre-bootstrap service as retryable, bootstraps the plist, retries enable, and verifies with launchctl print. Other enable errors still fail before bootstrap.
